sites/kit.svelte.dev/src/lib/search/SearchBox.svelte

```diff
@@ -65,7 +64,13 @@
 	}
 
 	function update() {
-		results = (index ? index.search($query) : []).map((href) => lookup.get(href));
+		const filter = /type|migrat/;
+		let searchResults = index ? index.search($query) : [];
+		// only include type and migration docs if user is explicitly searching for them
+		if (!filter.test($query.toLowerCase())) {
+			searchResults = searchResults.filter((href) => !filter.test(href));
+		}
+		results = searchResults.map((href) => lookup.get(href));
```

Comment:
```suggestion
		const requestedTypesOrMigrating = filter.test($query.toLowerCase())
		results = searchResults
			.map((href) => lookup.get(href))
			.filter(result => requestedTypesOrMigrating || !filter.test(result.breadcrumbs[0].toLowerCase()));
```
I might've messed up the boolean algebra (blame github's editor), but this should have fewer false negatives if I didn't

Reply:
Thanks! That definitely is a better approach 🥰 
But there's actually a bug in this: breadcrumbs might be of length 0 because it is only a sliced version of the original breadcrumbs.
I've also mashed everything into one big reduce to not have to iterate over the array twice